From: Detlev Casanova <detlev.casanova@collabora.com>
Date: Wed, 25 Jan 2023 11:44:31 -0500
Subject: upcm: Build it as a module

the upcm module doesn't behave properly when built-in. It reports an
invalid argument when trying to set the channel count, resulting in
audio_server closing.

Signed-off-by: Detlev Casanova <detlev.casanova@collabora.com>
---
 drivers/ks-t1/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/ks-t1/Kconfig b/drivers/ks-t1/Kconfig
index ec01331..d5bccc8 100644
--- a/drivers/ks-t1/Kconfig
+++ b/drivers/ks-t1/Kconfig
@@ -1,10 +1,10 @@
 menu "KS-T1 drivers"
 
 config ALSA_UPCM
-	bool "ALSA UPCM driver"
+	tristate "ALSA UPCM driver"
 	depends on ARCH_ROCKCHIP
 	select SWITCH
-	default y
+	default m
 	help
 	  Build the upcm driver for the KS-T1 board
 
